### DISCUSSION TITLE
Filter all occurrences between 2 matches

### DISCUSSION THREAD

**Discussion by lucasbracher** [zeroSteiner/rule-engine#50] [Q&A] [answered]
Hello, all!

Let's say I have the following list:

`l = [{'name': 'a'}, {'name': 'b'}, {'name': 'c'}, {'name': 'd'}, {'name': 'e'}]`

I would like to create a filter that returns all occurrences between `{'name': 'a'}`, including it and `{'name': 'd'}`, excluding it.

Is it possible?

Best regards,

Lucas Bracher

**Comment by zeroSteiner** [comment, marked as answer]
I can't think of a way to do it off-hand. You would need something like an index function.

**Reply by lucasbracher** [reply]
Oh, I see! Thanks for this idea!

Do you think it would be a nice feature to be included in the language? I can implement it and submit the MR to you.
I thought it could be something like this:

`'between first_name == "Luke" inclusive and email =~ ".*@rebels.org$" exclusive'`

`and` operator should be mandatory in this case.
The default behavior is to stop at the first match for both operands.
This is the idea, in a nutshell.

What do you think about it?

**Reply by zeroSteiner** [reply]
If it were implemented as a function instead of a special keyword, it'd be alot more useful. There's a ticket for #32 already to add functions, then you'd just need something like `index(container, item)` that would let you do something along the lines of `my_list[index(my_list, 'a'):index(my_list, 'b')]`.

Adding functions is a good idea, it would just be alot of work for me. Adding special keywords like this solution proposes would be more readable, but it might be more difficult to parse and probably wouldn't be too much easier to implement than functions.
